types/koa-generic-session/index.d.ts

```diff
@@ -0,0 +1,52 @@
+// Type definitions for koa-generic-session 1.x
+// Project: https://github.com/koajs/generic-session
+// Definitions by: Nick Simmons <https://github.com/nsimmons/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as Koa from "koa";
+
+declare namespace koaSession {
+    interface Session {
+        cookie: any;
+    }
+
+    interface SessionIdStore {
+        get(): any;
+        set(sid: string, session: Session): void;
+        reset(): void;
+    }
+
+    interface SessionStore {
+        (): SessionStore;
+        get(sid: string): any;
+        set(sid: string, session: Session, ttl: number): void;
+        destroy(sid: string): void;
+    }
+
+    interface SessionOptions {
+        key?: string;
+        store?: SessionStore;
+        ttl?: number;
+        prefix?: string;
+        cookie?: {
+            path?: string;
+            rewrite?: boolean;
+            signed?: boolean;
+            maxAge?: number;
+            secure?: boolean;
+            httpOnly?: boolean;
+        };
+        defer?: boolean;
+        rolling?: boolean;
+        sessionIdStore?: SessionIdStore;
+        genSid?(length: number): string;
+        errorHandler?(error: Error, type: string, ctx: Koa.Context): void;
+        valid?(ctx: Koa.Context, session: Session): boolean;
+        beforeSave?(ctx: Koa.Context, session: Session): void;
```

Comment:
Don't you need  below as members of options?
- allowEmpty : allow generation of empty sessions
- reconnectTimeout : When store is disconnected, don't throw  store unavailable  error immediately, wait  reconnectTimeout  to reconnect, default is  10s .